core/src/conn.rs

```diff
@@ -2,11 +2,12 @@ use crate::odbc_uri::UserOptions;
 use crate::{err::Result, Error};
 use crate::{MongoQuery, TypeMode};
 use bson::{doc, Bson, UuidRepresentation};
-use mongodb::sync::Client;
+use mongodb::Client;
 use serde::{Deserialize, Serialize};
 use std::time::Duration;
 
 #[derive(Debug)]
+#[repr(C)]
 pub struct MongoConnection {
     /// The mongo DB client
     pub client: Client,
```

Comment:
@nbagnard this will become `Arc<Client>` for OIDC, and somehow I will map ODBCUri to `Arc<Client>` so that we can retrieve the same client for anything using the same attributes.

Reply:
Actually! `Client` already uses an `Arc`, so all we have to do is clone it, no change to anything. It's actually an intended part of the API design to share `Client`s between async tasks and threads.